gsi/proxy/proxy_core/source/library/globus_gsi_proxy.c

```diff
@@ -136,7 +136,14 @@ globus_l_gsi_proxy_activate(void)
         CRYPTO_malloc_debug_init();
         CRYPTO_dbg_set_options(V_CRYPTO_MDEBUG_ALL);
 #endif
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
+/* From CRYPTO_mem_ctrl.3ossl man page: The memory-leak checking has been
+ * deprecated in OpenSSL 3.0 in favor of clang's memory and leak sanitizer.
+ * Alternatively, openssl needs to be configured using
+ *  ./Configure enable-crypto-mdebug
+ */
         CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_ON);
```

Comment:
@msalle:
`CRYPTO_mem_ctrl(CRYPTO_MEM_CHECK_ENABLE);` is "missing" here, compared to the test case above. From https://github.com/openssl/openssl/blob/b8be229dab036b26de8830444bf2beb82e71f50e/include/openssl/crypto.h.in#L349..L352 I can't make up what the "correct" sequence is if a sequence is needed at all.

Reply:
I agree it's not clear and I think I'd prefer to just not touch that part since it's unrelated to OpenSSL3.0 fixing.

Reply:
Agreed.